pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,284 @@
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	encryptionconfiguration "github.com/gardener/gardener/pkg/operation/etcdencryption"
+	"github.com/gardener/gardener/pkg/utils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+/*
+CreateOrUpdateEtcdEncryptionConfiguration creates or updates a secret on the seed
+which contains the EncryptionConfiguration that is necessary to encrypt the
+kubernetes secrets in etcd.
+
+More clearly, the secret conitself contains in its data element the EncryptionConfiguration
+as YAML encryption-configuration.yaml. The helm chart of the apiserver deployment mounts the
+secret to the apiserver and sets a new start parameter
+--encryption-provider-config=/etc/kubernetes/etcd-encryption-secret/encryption-configuration.yaml
+thereby making the EncryptionConfiguration known to the apiserver and triggering
+(depending on whether the EncryptionConfiguration was "active") the encryption of the
+kubernetes secrets in etcd.
+
+A second copy of the secret is stored in Gardener.
+
+Once the configuration is created and has become active (and the shoot's apiservers have been
+restarted), the second task RewriteShootSecretsIfEncryptionConfigurationChanged will trigger
+a rewrite of the shoot's secrets to ensure that the current encryption setting is taken
+into account.
+*/
+func (b *Botanist) CreateOrUpdateEtcdEncryptionConfiguration(ctx context.Context) error {
+	ecSeed, forcePlaintextSecrets, err := b.readEncryptionConfigurationFromSeed(ctx)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		b.Logger.Info("no EncryptionConfiguration found in seed. Creating new passive configuration.")
+		ecSeed, err = encryptionconfiguration.NewPassiveConfiguration()
+		if err != nil {
+			return err
+		}
+		return b.updateEtcdEncryptionConfiguration(ctx, ecSeed, forcePlaintextSecrets, true)
+	}
+	// The etcd encryption configuration does already exist, let's first check whether it is consistent
+	err = encryptionconfiguration.Validate(ecSeed)
+	if err != nil {
+		return fmt.Errorf("EncryptionConfiguration in seed cluster is not consistent: %v", err)
+	}
+	// Ensure the seed EncryptionConfiguration is equal to the one that was replicated to the
+	// project namespace in the garden cluster.
+	ecGarden, err := b.readEncryptionConfigurationFromGarden(ctx)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("EncryptionConfiguration exists in seed but not in garden")
```

Comment:
1. Errors should start with lowercase letters. This also applies to all other errors.
2. Make sure to include the error via `%v`

Reply:
yes, one of the past 200+ comments already suggested to start the errors in lowercase which I tried to do. 

I would have thought this does not apply to "standing names" such as the camelcase EncryptionConfiguration which I always write "EncryptionConfiguration" regardless whether at the beginning or not. 

I left out the not found error on purpose since I write in my message more context information (exists in one place but not the other) ...

Reply:
I changed the error and log messages to always start with lowercase ...